src/main/java/org/gbif/doi/service/datacite/RestJsonApiDataCiteService.java

```diff
@@ -0,0 +1,206 @@
+package org.gbif.doi.service.datacite;
+
+import com.github.jasminb.jsonapi.JSONAPIDocument;
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.NotImplementedException;
+import org.gbif.api.model.common.DOI;
+import org.gbif.api.model.common.DoiData;
+import org.gbif.api.model.common.DoiStatus;
+import org.gbif.datacite.model.json.Datacite42Schema;
+import org.gbif.datacite.rest.client.DataCiteClient;
+import org.gbif.datacite.rest.client.model.DoiSimplifiedModel;
+import org.gbif.datacite.rest.client.model.EventType;
+import org.gbif.doi.metadata.datacite.DataCiteMetadata;
+import org.gbif.doi.service.DoiException;
+import org.gbif.doi.service.DoiService;
+import retrofit2.Response;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.Base64;
+
+/**
+ * DataCite service implementation with REST and JSON:API.
+ */
+public class RestJsonApiDataCiteService implements DoiService {
+
+    private DataCiteClient dataCiteClient;
+
+    public RestJsonApiDataCiteService(DataCiteClient dataCiteClient) {
+        this.dataCiteClient = dataCiteClient;
+    }
+
+    /**
+     * Resolves the registered identifier to its status and target URL.
+     * @param doi the identifier to resolve
+     * @return the status object with the target URL the DOI is backed by or null if DOI does not exist at all
+     */
+    @Nullable
+    @Override
+    public DoiData resolve(DOI doi) {
+        Preconditions.checkNotNull(doi);
+
+        Response<JSONAPIDocument<Datacite42Schema>> doiResponse = dataCiteClient.getDoi(doi.getDoiName());
+        JSONAPIDocument<Datacite42Schema> bodyJsonApiWrapper = doiResponse.body();
+
+        if (!doiResponse.isSuccessful() || bodyJsonApiWrapper == null) {
+            return new DoiData(DoiStatus.FAILED, null);
+        }
+
+        Datacite42Schema body = bodyJsonApiWrapper.get();
+        String doiState = body.getState();
+
+        if ("draft".equals(doiState)) {
```

Comment:
maybe we should add a constructor or factory method to  DoiData(DoiStatus) to avoid those explicit null

Reply:
added to gbif-api